interfaces/devdoc/sync_requirements.md

```diff
@@ -0,0 +1,42 @@
+# sync
+================
+
+## Overview
+
+`sync` provides platform-independent synchronization primitives.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout);

```

Comment:
>timeout [](start = 115, length = 7)

This should be `timeout_milliseconds` or some descriptive name, here and everywhere